### PR TITLE
Fix linting issues found by clang-tidy 6.0

### PR DESCRIPTION
**Pull request by mm318**
*Description of changes:*

Fixing linting issues found by `clang-tidy` 6.0, without introducing public API changes (there are ABI changes though).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws_common/src/sdk_utils/auth/service_credentials_provider.cpp [hunk 1]
```diff
@@ -66,7 +65,7 @@ static bool SetCurlOpt(CURL * curl, CURLoption opt, T lvalue);
 static bool AppendHeader(struct curl_slist ** headers, const char * name, const char * value);
 static bool IsIotConfigValid(const IotRoleConfig & config);
 static bool ParseConfigVale(std::string & value, Aws::String & result);
-static bool ParseConfigValue(std::string & value, long & result);
+static bool ParseConfigValue(std::string & value, int & result);
```
Comment (AAlon) [inline aws_common/src/sdk_utils/auth/service_credentials_provider.cpp]: Why?
Can we use int32_t instead?

File: aws_common/src/sdk_utils/auth/service_credentials_provider.cpp [hunk 2]
```diff
@@ -310,17 +314,16 @@ class RequestContext
 // IotRoleCredentialsProvider Class
 //
 
-IotRoleCredentialsProvider::IotRoleCredentialsProvider(const IotRoleConfig & config)
+IotRoleCredentialsProvider::IotRoleCredentialsProvider(const IotRoleConfig & config) : expiry_(0.0)
```
Comment (AAlon) [inline aws_common/src/sdk_utils/auth/service_credentials_provider.cpp]: Moving only one of them to the initializer list looks weird

File: aws_common/src/sdk_utils/auth/service_credentials_provider.cpp (hunk 1 above)
Comment (mm318) [reply aws_common/src/sdk_utils/auth/service_credentials_provider.cpp]: The [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Integer_Types) discourages the use of `short`, `long`, `long long`, etc. Using `int` is acceptable.

Sure, using `int32_t` is a good idea.

Comment (mm318) [reply aws_common/src/sdk_utils/auth/service_credentials_provider.cpp]: Done.

File: aws_common/src/sdk_utils/auth/service_credentials_provider.cpp (hunk 2 above)
Comment (mm318) [reply aws_common/src/sdk_utils/auth/service_credentials_provider.cpp]: Done.

**Pull request merged**